src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/grid.controller.js

```diff
@@ -343,6 +343,37 @@ angular.module("umbraco")
             }
         };
 
+        var shouldApply = function(item, itemType, gridItem) {
+            if (item.applyTo === undefined || item.applyTo === null || item.applyTo === "") {
+                return true;
+            }
+
+            if (typeof (item.applyTo) === "string") {
+                return item.applyTo === itemType;
+            }
+
+            if (itemType === "row") {
+                if (item.applyTo.row === undefined) {
```

Comment:
This is now the "json configuration" logic which works with the json object:

Undefined means the row property has been left out completely from the json config object. This is like when you would use only "cell" value for the old applyTo selector ... simply not defining a row will disallow it (return false).